bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4283,6 +4286,18 @@ ReplicaImp::ReplicaImp(bool firstTime,
 
   KeyExchangeManager::instance(&id);
 
+  DbCheckpointManager::Instance().addOnStableSeqNum([this](const SeqNum &seqNum) {
+    auto currTime = std::chrono::system_clock::now().time_since_epoch();
+    auto timeSinceLastSnapshot = (std::chrono::duration_cast<std::chrono::seconds>(currTime) -
+                                  DbCheckpointManager::Instance().getLastCheckpointCreationTime())
+                                     .count();
+    if (getReplicaConfig().maxNumberOfDbCheckpoints && seqNum &&
+        !(seqNum % getReplicaConfig().dbCheckPointWindowSize) &&
+        (timeSinceLastSnapshot >= getReplicaConfig().dbSnapshotIntervalSeconds.count())) {
+      DbCheckpointManager::Instance().onCreateDbCheckpoint(seqNum);
```

Comment:
It's essentially no `onCreateDbCheckpoint()`, but `createDbCheckpoint()`

Reply:
fixed